app/controllers/user_controller.rb

```diff
@@ -20,30 +20,30 @@ def signup
   def update_user
     current_user = User.find(params[:id])
     payload = JSON.parse(request.body.read)
-    current_user.first_name = payload["first_name"] unless payload['first_name']&.empty?
-    current_user.email = payload["email"] unless payload['email']&.empty?
-    current_user.last_name = payload["last_name"] unless payload['last_name']&.empty?
-    current_user.user_type = payload["user_type"] unless payload['user_type']&.empty?
-
-    if current_user.valid?
-      current_user.save
+    update_detail = custom_compact(payload)
+    begin
+      current_user.update!(update_detail)
       render json: current_user, status: :ok
-    else
-      render json: current_user.errors.details, status: :bad_request
+    rescue => exception
+      render json: exception, status: :bad_request
     end
   end
 
   def show
     user = User.find_by(:id => params[:id])
     if user.present?
-      render json: { data: user}, status: :ok
+      render json: { data: user }, status: :ok
     else
       render json: { message: "User with Id: " + params[:id] + " is not found" }, status: :not_found
     end
   end
 
   private
 
+  def custom_compact(payload)
+    payload.select { |_, value| !value.empty? }
```

Comment:
👍

Alternatively, you could also push this responsibility to the front end to make sure we don't push anything null to the server-side unless we really meant to set it to null.

Reply:
By the way, there is a method called `#reject` which is the opposite of `#select`, so you could simplify this line slightly to:

```rb
payload.reject { |_, value| value.empty? }
```